### PR TITLE
[Feature][Plugin] Add plugin-spark-source-hdfs into v2

### PR DESCRIPTION
**Pull request by zixi0825**
---

## Purpose of this pull request

This pull request add spark hdfs source plugin into v2

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

File: plugin-spark-source-hdfs/pom.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>io.github.interestinglab.seatunnel</groupId>
+            <artifactId>plugin-spark-source-file</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>io.github.interestinglab.seatunnel</groupId>
+            <artifactId>seatunnel-spark-api</artifactId>
```
Comment (leo65535) [inline plugin-spark-source-hdfs/pom.xml:41]: hi, please rebase `dev` branch, we already change to `seatunnel-api-spark`

**Comment by timothy1024** [comment]
hi @CalvinKirs, seems the `Dead Link Checker` is unstable sometime. 

```
FILE: ./README_zh_CN.md
  [✖] https://www.qutoutiao.net/

  19 links checked.
  [✖] https://www.qutoutiao.net/ → Status: 0

  ERROR: 1 dead links found!
Error: Process completed with exit code 1.
```

**Comment by CalvinKirs** [comment]
> https://www.qutoutiao.net/

thx,part of the link is unstable, let's ignore it for now. I will concentrate on sorting out and repairing it later.

**Comment by zixi0825** [comment]
cc @CalvinKirs

File: plugin-spark-source-hdfs/pom.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>seatunnel</artifactId>
+        <groupId>io.github.interestinglab.seatunnel</groupId>
```
Comment (simon824) [inline plugin-spark-source-hdfs/pom.xml:25]: let's move `artifactId` under `groupId`.

**Comment by CalvinKirs** [comment]
We already have the `plugin-spark-source-file` module. Do we still need this?

**Comment by zixi0825** [comment]
> We already have the `plugin-spark-source-file` module. Do we still need this?

easy to use, don not need to write prefix schema hdfs://, the sink also has the file and hdfs

**Comment by CalvinKirs** [comment]
> > We already have the `plugin-spark-source-file` module. Do we still need this?
> 
> easy to use, don not need to write prefix schema hdfs://, the sink also has the file and hdfs

If it is just this, it seems that there is no substantial improvement for users. Let us hear what other people think.

**Comment by zixi0825** [comment]
> > > We already have the `plugin-spark-source-file` module. Do we still need this?
> > 
> > 
> > easy to use, don not need to write prefix schema hdfs://, the sink also has the file and hdfs
> 
> If it is just this, it seems that there is no substantial improvement for users. Let us hear what other people think.

I move the hdfs implement into the file module, do not need the independent module

**Pull request closed without merging**